File: javascript/mumbling/src/mumbling.js
class Mumbling {

  convert(input) {
    if (input === "") return "";
    let mumble = "";
    input.split("").forEach((char, index) => {
      mumble += this.convertChar(char.toLowerCase(), index);
      if (index < input.length - 1) mumble += "-";
    });
    return mumble;
  }

  convertChar(char, index) {
    let returnString = "";
    for (let i = 0; i < index + 1; i++) {
      if (i === 0) {
        returnString += char.toUpperCase();
      } else {
        returnString += char;
      }
    }
    return returnString;
  }
}

module.exports = Mumbling;

File: python/mumbling/mumbling.py
class Mumbling:

  def __init__(self):
    pass

  def mumble(self, str):
    if len(str) == 0: return str
    if len(str) == 1: return str.upper()
    output = str[0].upper()
    for i in range(1, len(str)):
      output += "-" + self.convert_char(str[i], i + 1)
    return output;


  def convert_char(self, letter, index):
    returnStr = letter.upper()
    letter = letter.lower()
    for i in range(index - 1):
      returnStr += letter
    return returnStr